syntax/attribute_value.ml

```diff
@@ -484,6 +484,33 @@ let paint ?separated_by:_ ?default:_ loc name s =
             `Icc ([%e iri], Some [%e paint_without_icc loc name remainder])]
     end [@metaloc loc]
 
+let opacity =
+  let bad_form name loc =
+    Common.error loc "Value of %s must be a number or percentage" name in
+
+  let regexp = Re_str.regexp "\\([-+0-9eE.]+\\)\\(%\\)?" in
+
+  fun ?separated_by:_ ?default:_ loc name s ->
+    if not @@ does_match regexp s then bad_form name loc;
+
+    begin
+
+      try
+        let n = float_of_string (Re_str.matched_group 1 s) in
+
+        let v =
+          if group_matched 2 s then (n /. 100.)
+          else n in
+
+        if v >= 0. && v <= 1. then
+          Some [%expr [%e (Common.float loc @@ v)]]
+        else
+          Common.error loc "Value of %s must be between 0 and 1." name
+
+      with Failure _ -> bad_form name loc
+
+    end [@metaloc loc]
```

Comment:
That is a very spacious formatting :smile:

Reply:
I tried to match the style of the rest of the file ☺ I agree it's quite spaced, and I'm fine with changing this.